From b4fafb6d6afbdf3f2c5859fec8300576d73a8552 Mon Sep 17 00:00:00 2001
From: Renjaya Raga Zenta <ragazenta@gmail.com>
Date: Wed, 18 Dec 2024 16:44:32 +0700
Subject: [PATCH] overlays: goodix: Allow override i2c address

Some Goodix devices e.g. gt911 use address 0x5d instead of 0x14.
So, make the address overridable.

Signed-off-by: Renjaya Raga Zenta <ragazenta@gmail.com>
---
 arch/arm/boot/dts/overlays/README             | 3 ++-
 arch/arm/boot/dts/overlays/goodix-overlay.dts | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1454,7 +1454,8 @@ Name:   goodix
 Info:   Enables I2C connected Goodix gt9271 multiple touch controller using
         GPIOs 4 and 17 (pins 7 and 11 on GPIO header) for interrupt and reset.
 Load:   dtoverlay=goodix,<param>=<val>
-Params: interrupt               GPIO used for interrupt (default 4)
+Params: addr                    I2C address (default 0x14)
+        interrupt               GPIO used for interrupt (default 4)
         reset                   GPIO used for reset (default 17)
         i2c-path                Override I2C path to allow for i2c-gpio buses
 
--- a/arch/arm/boot/dts/overlays/goodix-overlay.dts
+++ b/arch/arm/boot/dts/overlays/goodix-overlay.dts
@@ -37,6 +37,7 @@
 	};
 
 	__overrides__ {
+		addr = <&gt9271>,"reg:0";
 		interrupt = <&goodix_pins>,"brcm,pins:0",
 			<&gt9271>,"interrupts:0",
 			<&gt9271>,"irq-gpios:4";
